### PR TITLE
Form

### PR DESCRIPTION
**Pull request by iambenkis**
In this milestone, I create forms in my Blog App.

### Checklist :1st_place_medal: 

-  Create a method called `current_user` in your ApplicationController to make current user data available to all controllers. :heavy_check_mark: 
    -  It will return the first user from the database.
    -  Don't worry about proper authentication at this time. You will add it in the upcoming projects.
- Create forms to perform the following functions: :heavy_check_mark: 
    - Creates a Post on behalf of the `current_user` (use the method that you created in your ApplicationController).
    - Create a comment on behalf of the `current_user` (use the method that you created in your ApplicationController).
- Allow Users to add likes to posts. :heavy_check_mark:

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -7,6 +7,44 @@ def index
...
+    @comment = Comment.new(comment_params)
+    @comment.post = Post.find(params[:id])
+    @comment.author = current_user
+    redirect_to post_path if @comment.save
+  end
+
+  def create_like
+    @like = Like.new(author: current_user, post: Post.find(params[:id]))
+    redirect_to post_path if @like.save
+  end
```
Comment (rloterh) [inline app/controllers/posts_controller.rb:24]: - [ ] In this activity, we want you to place the like actions (`create`) in a _likes_controller_, to enforce separation of concerns principle. This should be similar to:

```ruby
# controllers/likes_controller.rb

  class LikesController < ApplicationController
    ...

    def create
      @post = Post.find(params[:post_id])
      @like = @current_user.likes.new(post_id: @post.id)

      respond_to do |format|
        if @like.save
          format.html { redirect_to @post }
        else
          format.html { redirect_to @post }
        end
      end
    end
	  ...

  end
```

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -7,6 +7,44 @@ def index
...
+    @comment = Comment.new
+    @like = Like.new
+  end
+
+  def create_comment
+    @comment = Comment.new(comment_params)
+    @comment.post = Post.find(params[:id])
+    @comment.author = current_user
+    redirect_to post_path if @comment.save
+  end
```
Comment (rloterh) [inline app/controllers/posts_controller.rb:19]: - [ ] Simiarly, comments action (`new` &`create`) should be defined in _comments_controller_, to enforce separation of concerns principle. This should be similar to:

```ruby
# controllers/comments_controller.rb

class CommentsController < ApplicationController
  ...

  def new
    @comment = Comment.new
  end

	def create
    @comment = @current_user.comments.new(params)
    @comment.post_id = params[:post_id]

    if @comment.save
      redirect_to post_path(@comment.post)
    else
      @article = Post.find(params[:post_id])
      redirect_to post_path(@post)
    end
  end

  private 
  def comment_params
    params.require(:comment).permit(:text)
  end

	...
end
```

File: config/routes.rb [hunk 3]
```diff
@@ -4,9 +4,14 @@
   # Defines the root path route ("/")
   # root "articles#index"
   root "users#index"
+
+  get '/posts/new', to: 'posts#new', as: 'create_post'
+  post '/posts/new', to: 'posts#create'
+  get '/users/:user_id/posts/:id', to: 'posts#show', as: 'post'
+  post '/users/:user_id/posts/:id/', to: 'posts#create_comment'
+  post '/users/:user_id/posts/:id/likes', to: 'posts#create_like', as: 'like'
+
```
Comment (rloterh) [inline config/routes.rb:13]: - [ ] Kindly ensure that there is a pair of routes in the `config/routes.rb` file which corresponds with the `new` and `create`, actions in the comments controller. Viewing these routes in the command line should be similar to:
```ruby
   POST   /posts/:post_id/comments(.:format)     comments#create
   GET   /posts/:post_id/comments/new(.:format)     comments#new

Comment (rloterh) [inline config/routes.rb:13]: - [ ] Similarly, kindly ensure that there is a route in the `config/routes.rb` file that corresponds with the `create` actions in the likes_controller. This route in the command line should be similar to:
```ruby
  `POST /posts/:post_id/likes(.:format) likes#create`

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Benjamin, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦 🚦 🚦  

## Highlights 👍  
- Nice job using strong params in the _posts_controller_

## Required Changes ♻️
_Kindly check the inline comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

**You can also consider:**
- Kindly endeavor to fill the FAQ section with at least 2 Q&A(s).
- MIT license link in your ReadMe is broken, kindly endeavor to fix it, and remove guide note below it.

Implementing the above change(s) will make your ReadMe/ repository stand out even better!😉

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: config/routes.rb (hunk 3 above)
Comment (iambenkis) [reply config/routes.rb:13]: Thank  you sir :pray:

**Review by Reem-lab — APPROVED** [review]
# Hi @iambenkis, 👋🏻👋🏻 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉🚀🔥🎉🎉 


## Highlights ✨✨
- Followed Gitflow ✅
- Great working on the functionality 🔥🚀🎉
- Descriptive PR ✅

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**